src/Catalyst.Core.Modules.P2P.Discovery.Hastings.Tests/UnitTests/DiscoveryTestBuilder.cs

```diff
@@ -125,9 +126,10 @@ public DiscoveryTestBuilder WithPeerSettings(IPeerSettings peerSettings = defaul
         {
             _dnsClient = dnsClient == default(IDns) && mock == false
                 ? Substitute.For<IDns>()
-                : DiscoveryHelper.MockDnsClient(_peerSettings = _peerSettings == null && peerSettings == default(IPeerSettings)
-                    ? PeerSettingsHelper.TestPeerSettings()
-                    : peerSettings);
+                : DiscoveryHelper.MockDnsClient(_peerSettings =
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Extract the assignment of '_peerSettings' from this expression.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4942504)